health_checks.js

```diff
@@ -0,0 +1,103 @@
+/* -----------------------------------------------------------------------------
+ * @copyright (C) 2018, Alert Logic, Inc
+ * @doc
+ *
+ * Halth check functions common for all collectors
+ *
+ * @end
+ * -----------------------------------------------------------------------------
+ */
+
+const AWS = require('aws-sdk');
+const async = require('async');
+
+/**
+ * @function
+ *
+ * @param {Object} event - checkin event for Lambda function, can be used to send needed parameters for health checks. Example: stackName.
+ * @param {Object} context - context of Lambda's function.
+ * @param {function} callback - callback, which is called by health check when it's done.
+ *
+ * @returns {(null|ErrorMsg)} returns null if all OK or ErrorMsg if something was wrong
+ */
+
+function checkCloudFormationStatus(event, context, callback) {
+    var stackName = event.StackName;
+    var cloudformation = new AWS.CloudFormation();
+    cloudformation.describeStacks({StackName: stackName}, function(err, data) {
+        if (err) {
+            return callback(errorMsg('ALAWS00001', stringify(err)));
+        } else {
+            var stackStatus = data.Stacks[0].StackStatus;
+            if (stackStatus === 'CREATE_COMPLETE' ||
+                stackStatus === 'UPDATE_COMPLETE' ||
+                stackStatus === 'UPDATE_IN_PROGRESS' ||
+                stackStatus === 'UPDATE_ROLLBACK_COMPLETE' ||
+                stackStatus === 'UPDATE_ROLLBACK_IN_PROGRESS' ||
+                stackStatus === 'UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS' ||
+                stackStatus === 'REVIEW_IN_PROGRESS') {
+                return callback(null);
+            } else {
+                return callback(errorMsg('ALAWS00002', 'CF stack has wrong status: ' + stackStatus));
+            }
+        }
+    });
+}
+
+function getHealthStatus(event, context, customChecks, callback) {
+    customChecksFuns = customChecks.map(f => function(asyncCallback) {
+        f(event, context, asyncCallback)
+    });
+    async.parallel([
```

Comment:
@kkuzmin we discussed `async.map` but I think `async.parallel` suits us better: 

https://caolan.github.io/async/docs.html#parallel

* Run the tasks collection of functions in parallel, without waiting until the previous function has completed. 
* If any of the functions pass an error to its callback, the main callback is immediately called with the value of the error. 

So it's basically the same behaviour as `waterfall`, so we don't even have to change interfaces

Reply:
@kkuzmin are you OK with `async.parallel`?